chia/consensus/blockchain.py

```diff
@@ -234,12 +234,15 @@ async def receive_block(
             None,
         )
         # Always add the block to the database
-        await self.block_store.add_full_block(block, block_record)
-
-        self.add_block_record(block_record)
-
-        fork_height: Optional[uint32] = await self._reconsider_peak(block_record, genesis, fork_point_with_peak)
-
+        fork_height = None
+        async with self.block_store.db_wrapper.lock:
+            try:
+                await self.block_store.add_full_block(block, block_record)
```

Comment:
Technically we don't need to lock while adding the full block, since that does not change the peak. But i don't think it's a big deal

Reply:
We need lock for all writes, otherwise it's possible to get either "Can't begin transaction in transaction" error or it's possible to commit to db before intended transaction is done.